lib/appmap/event.rb

```diff
@@ -92,7 +92,12 @@ def build_from_invocation(mc = MethodCall.new, defined_class, method, receiver,
             mc.parameters = method.parameters.map.with_index do |method_param, idx|
               param_type, param_name = method_param
               param_name ||= 'arg'
-              value = arguments[idx]
+
+              if (arguments.length == 1) && (arguments[0].is_a? Hash) && (arguments[0][param_name])
+                value = arguments[0][param_name]
+              else
+                value = arguments[idx]
+              end
```

Comment:
Hash isn't always the first argument, but it's always the last one. You should look at the parameter type instead, and use `arguments[-1]` if the type is `:key` or `:keyreq`. (Not sure what'll be in `arguments` if all keyword arguments are optional and none is provided; hopefully an empty hash in the last place. If not, you'll have to pop the posititonal arguments to determine if keyword arguments are provided.)

Reply:
It wasn't supposed to be solving a hash argument problem. It was a fix only for single edge case when params are collected it a wrong way. So I check it and fix on the fly. But thanks to your comment I examined it one more time and improved `if` condition to make it work more accurate.